test/diffeq.jl

```diff
@@ -14,6 +14,7 @@ u0 = LorenzVector(1.0,0.0,0.0)
 p = LorenzParameterVector(10.0,28.0,8/3)
 tspan = (0.0,10.0)
 prob = ODEProblem(f,u0,tspan,p)
+@test_broken sol = solve(prob,Rosenbrock23())
```

Comment:
why is this one broken? It should just use the StaticArray LU/QR dispatches?

Reply:
It is broken because ForwardDiff tries to do `vector_mode_jacobian` and it uses `MArray`, but `MArray` doesn't have labels.

Reply:
ForwardDiff doesn't internally use it as a SArray? And I'm surprised this doesn't use `similar` or `similar_type` to generate the internals, and thus keep the labels?

Reply:
```julia
julia> u0 |> typeof
SLArray{Tuple{3},1,(:x, :y, :z),Float64}

julia> similar(u0)
3-element MArray{Tuple{3},Float64,1,3}:
 3.3e-322
 6.9339732699493e-310
 0.0
```

Reply:
We need an `MArray` version to `similar` to then... there must be an easy way.